app/public/src/game/components/battle-manager.ts

```diff
@@ -427,7 +427,7 @@ export default class BattleManager {
               pkm.detail.hp.textContent = pokemon.life.toString()
             }
           } else if (change.field == "shield") {
-            if (change.value > 0) {
+            if (change.value >= 0) {
```

Comment:
this is the fix for the visual bug of shield remaining visible in life bars when at zero